src/background/modules/ActivityPubRedirect.js

```diff
@@ -0,0 +1,46 @@
+import * as NetworkTools from "/common/modules/NetworkTools.js";
+import * as AddonSettings from "/common/modules/AddonSettings/AddonSettings.js";
+import { resolveActivityPubId } from "/common/modules/MastodonApi.js";
+
+/**
+ * Scrapes the ActivityPub <link> destination from the HTML page, if needed and exists.
+ *
+ * @param {number} tabId
+ * @param {URL} url
+ * @returns {Promise}
+ */
+export async function redirectByActivityPubLink(tabId, url) {
+    if (!tabId || !url) {
+        throw new Error("Needs a tab id and a page URL");
+    }
+
+    const [objectId] = await browser.tabs.executeScript(tabId, {
+        code: `document.querySelector("link[rel=alternate][type='application/activity+json']")?.href`,
+        runAt: "document_end",
+    });
+    if (!objectId) {
+        return;
+    }
+
+    const ownMastodon = await AddonSettings.get("ownMastodon");
+    if (ownMastodon.server === url.hostname) {
+        return;
+    }
+
+    const body = await resolveActivityPubId(ownMastodon.server, objectId);
+
+    const baseUrl = `https://${ownMastodon.server}`;
+    let homeUrl;
+    if (body.accounts[0]) {
+        homeUrl = new URL(`@${body.accounts[0].acct}`, baseUrl);
+    } else if (body.statuses[0]) {
+        homeUrl = new URL(
+            `@${body.statuses[0].account.acct}/${body.statuses[0].id}`,
+            baseUrl
+        );
+    } else {
+        return;
+    }
+
+    await NetworkTools.redirectToWebsite(homeUrl, tabId);
```

Comment:
Hmm? I believe awaiting is more helpful, since an error will throw at await call site but not at return call site. And awaiting is anyway more consistent.

Reply:
hmm this goes deep into the way promises may be handled and I would have said before it makes no difference at all whether to await here, because likely where this is called it is also awaited. And await is just a cooler syntax for `then`.

But what you say may be right too, the stacktrace may not lost thus last function then? But I am really unsure here, so a js snippet/proof or some source would help. I am always open to learn something new hehe. Otherwise I'd stand to what I said, I.e. remove the await, because that would be consistent to how it is done in the code base everywhere else.